frontend/packages/ceph-storage-plugin/src/utils/time-duration-dropdown.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { Dropdown } from '@console/internal/components/utils';
+import { TextInput, ValidatedOptions } from '@patternfly/react-core';
+import { timeUnits } from '../constants/bucket-class';
+
+export const TimeDurationDropdown: React.FC<TimeDurationDropdownProps> = ({
+  inputClassName,
+  onChange,
+  required,
+  testID,
+  placeholder,
+  inputID,
+}) => {
+  const [unit, setUnit] = React.useState(timeUnits.HOUR);
+  const [value, setValue] = React.useState(1);
+  const [validated, setValidated] = React.useState(ValidatedOptions.success);
+  const onValueChange = (val) => {
+    setValue(val);
+    onChange({ value: val, unit }, setValidated);
+  };
+  const onUnitChange = (newUnit) => {
+    setUnit(newUnit);
+    onChange({ value, unit: newUnit }, setValidated);
+  };
```

Comment:
nit: add spaces above and below
```suggestion

  const onValueChange = (val) => {
    setValue(val);
    onChange({ value: val, unit }, setValidated);
  };
  const onUnitChange = (newUnit) => {
    setUnit(newUnit);
    onChange({ value, unit: newUnit }, setValidated);
  };
  
```

Reply:
fixed